odl/contrib/shearlab/shearlab_operator.py

```diff
@@ -0,0 +1,312 @@
+
+    def _call(self, x):
+        """Return ``self(x)``."""
+        with self.op.mutex:
+            x = np.moveaxis(x, 0, -1)
+            return shearrec2D(x, self.op.shearlet_system)
+
+    @property
+    def adjoint(self):
+        """Return the adjoint operator."""
+        return ShearlabOperatorAdjointInverse(self.op)
+
+    @property
+    def inverse(self):
+        """Return the inverse operator."""
+        return self.op
+
+
+class ShearlabOperatorAdjointInverse(odl.Operator):
+
+    """Adjoint of the inverse/Inverse of the adjoint of shearlet transform.
+    Should not be used independently.
+    See Also
+    --------
+    odl.contrib.shearlab.ShearlabOperator
+    """
+
+    def __init__(self, op):
+        """Initialize a new instance.
+        Parameters
+        ----------
+        op : `ShearlabOperator`
+            The operator which this should be the inverse of the adjoint of.
+        """
+        self.op = op
+        super(ShearlabOperatorAdjointInverse, self).__init__(
+            op.domain, op.range, True)
+
+    def _call(self, x):
+        """Return ``self(x)``."""
+        with self.op.mutex:
+            result = shearrecadjoint2D(x, self.op.shearlet_system)
+            return np.moveaxis(result, -1, 0)
+
+    @property
+    def adjoint(self):
+        """Return the adjoint operator."""
+        return self.op.inverse
+
+    @property
+    def inverse(self):
+        """Return the inverse operator."""
+        return self.op.adjoint
+
+# Python library for shearlab.jl
+
+
+# Function to load Shearlab
+def load_Shearlab():
+    # Importing base
+    j = julia.Julia()
+    j.eval('using Shearlab')
+    j.eval('using PyPlot')
+    j.eval('using Images')
+    return j
+
+
+# Load Shearlab
+j = load_Shearlab()
+
+
+# Function to load images with certain size
+def load_image(name, n, m=None, gpu=0, square=0):
+    if m is None:
+        m = n
+    command = 'Shearlab.load_image(' + '"' + name + '"' + ',' + str(n) + ','
+    command = command + str(m) + ',' + str(gpu) + ',' + str(square) + ')'
+    return j.eval(command)
+
+
+# Function to plot images
+def imageplot(f, str='', sbpt=[]):
+    """
+        Use nearest neighbor interpolation for the display.
+    """
+    if sbpt != []:
+        plt.subplot(sbpt[0], sbpt[1], sbpt[2])
+    imgplot = plt.imshow(f, interpolation='nearest')
+    imgplot.set_cmap('gray')
+    pylab.axis('off')
+    if str != '':
+        plt.title(str)
+
+
+# Class of shearlet system in 2D
+class Shearletsystem2D:
+    def __init__(self, shearlets, size, shearLevels, full, nShearlets,
+                 shearletIdxs, dualFrameWeights, RMS, isComplex):
+        self.shearlets = shearlets
+        self.size = size
+        self.shearLevels = shearLevels
+        self.full = full
+        self.nShearlets = nShearlets
+        self.shearletIdxs = shearletIdxs
+        self.dualFrameWeights = dualFrameWeights
+        self.RMS = RMS
+        self.isComplex = isComplex
```

Comment:
Ok, I did classes since that is the style in julia when doing objects. The so called duck typing, this makes the julia code more efficient for the precompiler.

Reply:
I see. Well, duck typing is also a big topic in Python, but without the precompiler :-)
Not that it makes any difference here, but tuples tend to be more compact and (perhaps) faster. If I would have had to write this code from scratch, I would have used a dictionary. But it's up to you.

Reply:
I would like to keep this format, it does not affect the performance of the module, and it feels more easy to call, I also think this are attributes of a ShearletSystem that should be used in the future in the implemenation

Reply:
Okay, fair enough. But then please inherit from `object`, for our friends who are stuck in the past (Python 2).

Reply:
Holgers comment here should be adressed